src/array.vala

```diff
@@ -2,6 +2,9 @@ using GLib;
 
 public class Vast.Array : Object
 {
+    public const ssize_t NEW_AXIS = ssize_t.MAX;
+    public const size_t DEFAULT_DIMENSION = size_t.MAX;
```

Comment:
First, there is no insertion -- `axis` assigns / routes the axis to an axis of the original array or a newaxis. The dimension of the builder is already set when it is created. (`.builder(5)`).

I have been considering using transpose too -- a few drawbacks such that I didn't do it:

- The equivalent of `[:, newaxis, ...]` would be `arr.redim(arr.dimension + 1).transpose({0, dimension + 1, 1, 2, ...})`. It is a lot less direct -- I wanted to insert an axis but I have to do it via a transpose -- and create a full permutation array whatever I want to do?


- On the other hand, implementing `transpose` with `axis` is easy.

- In cases I want to swap two axis, `.transpose({1, 0, 2, 3, 4 ...})` I need to create redundant entries in the argument. compare this to current way, `.axis(0, 1).axis(1, 0)`. 

- thus I concluded, routining/assignment of axis (`axis`) is more fundamental than `.transpose`.

- I don't think the current function name `axis` correctly describes what the function does.